### ISSUE TITLE
Disable warning "Extra tokens on pragma line"

### ISSUE THREAD

**Issue by carlosgalvezp** [include-what-you-use/include-what-you-use#1207]
Hi,

Often times we want to use a `pragma` together with some short justification as to why we use it. For example:

```
#include "foo.h"  // IWYU pragma: keep (for Foo)
```

This currently leads to IWYU emitting a warning: `warning: Extra tokens on pragma line`. This is troublesome as it pollutes the logs.

Is there a way to achieve this use case without a warning? If not, could support be added to opt-out?

Thanks!

**Comment by carlosgalvezp** [comment]
Agreed! I believe most useful is to detect typos in the pragma. But if the pragma is correct I guess the tool could ignore what comes after.

**Comment by carlosgalvezp** [comment]
I see the code looks like this:

```
  if (tokens.size() > num_expected_tokens &&
      !StartsWith(tokens[num_expected_tokens], "//") &&
      !StartsWith(tokens[num_expected_tokens], "*/")) {
    Warn(loc, "Extra tokens on pragma line");
  }
```

Could we add one more case for starting with `(`? That would solve my use case.

**Comment by kimgr** [comment]
Oh, but that's your answer right there. Don't use `(for Foo)`, use `// for Foo`.

On the other hand I think the warning is pointless, so I'll switch it over to a higher-verbosity debug log.

**Comment by carlosgalvezp** [comment]
Hmm yes, it's just awkward to have double nested comments:

```
// IWYU pragma: keep // for Foo
```

Making it into higher severity works too!

**Comment by kimgr** [comment]
Check out the PR ^

**Comment by carlosgalvezp** [comment]
Looks great, thanks for the quick fix! :)

**Comment by kimgr** [comment]
Good times! Merged to master.
